prediction_market_agent/agents/microchain_agent/microchain_agent.py

```diff
@@ -270,7 +286,9 @@ def save_agent_history(
 
     # Restore the system prompt to its initial state
     agent.history[0] = dict(role="system", content=initial_system_prompt)
-    long_term_memory.save_history(agent.history)
+    long_term_memory.save_history(
+        agent.history[-save_last_n if save_last_n else len(agent.history) :]
+    )
 
     # Restore the head system prompt
     agent.history[0] = head_system_prompt
```

Comment:
I'm missing a bit of microchain knowledge here
-> I understand the agent is able to edit its system prompt (see `UpdateMySystemPrompt`)
-> The above edits `agent.history[0]`
-> I don't follow why we need to restore the `system_prompt` inside `save_agent_history`

Reply:
Evan explained it in the docstring

![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/a2c02585-75f2-43e2-8750-3a8b03a3cca8)